pkg/network/server.go

```diff
@@ -509,7 +514,10 @@ func (s *Server) handleGetDataCmd(p Peer, inv *payload.Inventory) error {
 				msg = s.MkMsg(CMDBlock, b)
 			}
 		case payload.StateRootType:
-			return nil // do nothing
+			r := s.stateCache.Get(hash)
```

Comment:
But what's the point of caching it, really? We have all the states in the DB anyway and can answer this call using that data.

Reply:
Well, there are 3 types of hashes: state root itself, hash of `StateRoot` struct and hash of the block. I don't think it is worth to have 1 and 2 in the blockchain itself. `StateRoot` from arbitrary blocks can be queried together via `StateRoots` payload and `Inventory` is usually used for new payloads.

Reply:
Actually 4, there is also hash of the P2P payload. And it is completely unrelated to anything `Blockchain` should deal with IMO.